### PR TITLE
[SMALLFIX] close connection when rpc fails to avoid leaking connections

### PR DESCRIPTION
**Pull request by peisun1115**

### CODE REVIEW

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -321,6 +323,7 @@ protected synchronized InetSocketAddress getAddress() {
         throw Throwables.propagate(AlluxioException.fromThrift(e));
       } catch (TException e) {
         LOG.error(e.getMessage(), e);
+        mProtocol.getTransport().close();
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/AbstractClient.java]: Should we just call `disconnect()` here?

Comment (peisun1115) [inline core/common/src/main/java/alluxio/AbstractClient.java]: done

**Comment by calvinjia** [comment]
LGTM

**Comment by ifcharming** [comment]
LGTM

**Pull request merged**